docs/docsite/rst/network_declarative_intent.rst

```diff
@@ -0,0 +1,349 @@
+       enabled: yes
+
+       # Intended state
+       state: connected
+
+
+**Advantages**
+
+* As you can tell from the above example using the declaritive intent format results in a much cleaner task. Checking no longer required the clunky (find better term) ``wait_for``.
+* FIXME: TBD: Add details of why it's better to use the module from an internal code point of view
+* Can be combined with ``aggregate:`` to easily ensure state across multiple items.
+* FIXME: Q: Why else
+
+FIXME: Q: Any disadvantages?
+
+* not all modules support declaritive_intent - raise an feature request LINK
+
+
+Physical then configuration
+---------------------------
+
+**Overview**
+
+* No point making configuration changes if someone hasn't plugged in the cable - simple case
+
+  * Checking the routing between connection - not just plugged in, but a route exists to the correct location - avoid cabling errors
+
+.. code-block:: yaml
+
+   - name: FIXME
+
+FIXME: Example
+
+**When would this be useful**
+
+* fixme
+
+**When would this not be useful**
+
+* fixme
+
+Configuration then physical
+---------------------------
+
+FIXME: Q: What would this look like
+
+**Overview**
+
+
+.. code-block:: yaml
+
+   - name: FIXME
+
+**When would this be useful**
+
+* Physical then configuration
+
+**When would this not be useful**
+
+
+
+Checking state only
+-------------------
+
+FIXME: Q: What would this look like
+
+**Overview**
+
+
+.. code-block:: yaml
+
+   - name: FIXME
+
+**When would this be useful**
+
+* fixme
+
+**When would this not be useful**
+
+* fixme
+
+Configuration, cabling, check
+-----------------------------
+
+FIXME: Q: What would this look like
+
+**Overview**
+
+
+.. code-block:: yaml
+
+   - name: FIXME
+
+**When would this be useful**
+
+* fixme
+
+**When would this not be useful**
+
+* fixme
+
+Aggreate declaritive intent
+---------------------------
+
+FIXME: Q: What would this look like
+
+**Overview**
+
+
+.. code-block:: yaml
+
+   - name: FIXME
+
+**When would this be useful**
+
+* fixme
+
+**When would this not be useful**
+
+* fixme
+
+Link to network_aggregate_resources for more info
+
+Rolling back: dealing with failure
+----------------------------------
+
+Block & Rescue
+
+* Roll back configuration?
+* Send (slack) notification that "{{ port }}" isn't configured, check cabling
+
+FIXME: Q: What would this look like
+
+**Overview**
+
+
+.. code-block:: yaml
+
+   - name: FIXME
+
+**When would this be useful**
+
+* fixme
+
+**When would this not be useful**
+
+* fixme
+
+Implementation details
+======================
+
+The ``delay:`` option
+----------------------
+
+All declaritive intent modules support a ``delay:`` option. This represents the amount of time, in seconds, that Ansible will wait after setting declaritive configuration before checking the indended state. This pause is needed to allow the network device being configured to stablise, such as to allow handshake after bring up an interface.
+
```

Comment:
Default value of delay is 10 seconds

Reply:
Added "The default value is specified in the module's documentation."
Module documentation updated in https://github.com/ansible/ansible/pull/30344